redwood-compose/src/commonMain/kotlin/app/cash/redwood/compose/RedwoodComposition.kt

```diff
@@ -56,10 +63,37 @@ public fun <W : Any> RedwoodComposition(
 ): RedwoodComposition {
   view.reset()
 
+  val saveableStateRegistry = view.savedStateRegistry?.let { viewRegistry ->
+    val state = viewRegistry.consumeRestoredState()
+    val composeRegistry = SaveableStateRegistry(state) { value ->
+      if (value is SnapshotMutableState<*>) {
+        if (value.policy === neverEqualPolicy<Any?>() ||
+          value.policy === structuralEqualityPolicy<Any?>() ||
+          value.policy === referentialEqualityPolicy<Any?>()
+        ) {
```

Comment:
We should probably have something similar in `StateSnapshot.kt` when serializing `MutableState`. Currently, we ignore the `policy` when serializing, and when deserializing, just set the `policy` to `structuralEqualityPolicy`.

Reply:
Yeah this is derived from Compose UI internals.